src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/ThunkPool.cs

```diff
@@ -347,19 +310,12 @@ public static unsafe IntPtr GetNewThunksBlock()
             }
             else
             {
-                nextThunksBlock = RuntimeImports.RhAllocateThunksMapping();
-
-                if (nextThunksBlock == IntPtr.Zero)
-                {
-                    // We either ran out of memory and can't do anymore mappings of the thunks templates sections,
-                    // or we are using the managed runtime services fallback, which doesn't provide the
-                    // file mapping feature (ex: older version of mrt100.dll, or no mrt100.dll at all).
-
-                    // The only option is for the caller to attempt and recycle unused thunks to be able to
-                    // find some free entries.
-
-                    return IntPtr.Zero;
-                }
+                nextThunksBlock = IntPtr.Zero;
+                int result = RuntimeImports.RhAllocateThunksMapping(&nextThunksBlock);
+                if (result == HResults.E_OUTOFMEMORY)
+                    throw new OutOfMemoryException();
+                else if (result != HResults.S_OK)
+                    throw new NotSupportedException() { HResult = result };
```

Comment:
> HResult = result

If we are passing the HRESULT through, we may want to return `COR_E_PLATFORMNOTSUPPORTED` on the unmanaged side instead of generic E_FAIL.

Reply:
I ended up just not propagating the HRESULT. The idea was that maybe we could have more detail here, but we currently don't so it feels premature.